### DISCUSSION TITLE
Device streaming feature status

### DISCUSSION THREAD

**Discussion by TFleury** [Azure/azure-iot-sdk-csharp#1811] [General]
Device streaming feature is not present in latests package releases, and there is no updated preview package including it.
It seams the last is 1.29.0-preview-004.

Has this feature been aborted ? Can we rely on it ?

**Comment by abhipsaMisra** [comment]
Device streaming is a preview specific feature, and is thus available only on our preview nugets.

The latest version would be:
Microsoft.Azure.Devices.Client 1.32.0-preview-001
Microsoft.Azure.Devices 1.28.0-preview-001

Our preview packages don't have as frequent of an update frequency as our GA nugets do. Let us know if there is a particular fix/ feature from `master` you are looking for in `preview`, we can get a plan in for a `preview` release then.

Adding @drwill-ms to comment on the future of Device Streaming feature.

**Comment by drwill-ms** [comment]
Device streaming is a feature not yet officially shipped by Azure IoT Hub as a "GA" (general audience) feature - it is still under preview. When it goes GA, we'll include it in our non-preview nugets.

I can't speak to when it'll be delivered, because plans continue to change based on competing priorities and customer requests. However, I know we've heard plenty of customer support for the feature.

To engage directly with the Azure IoT Hub team about feature requests, you can leave feedback (or support existing requests) at <https://feedback.azure.com/forums/321918-azure-iot-hub-dps-sdks>.

**Comment by drwill-ms** [comment]
Linking issue #986.

**Reply by drwill-ms** [reply]
Also #842.

**Comment by Marusyk** [comment]
It's a very required feature for our customers.
Especially in the West Europe region

@drwill-ms the feedback link doesn't work

> https://feedback.azure.com/forums/321918-azure-iot-hub-dps-sdks

**Reply by drwill-ms** [reply]
It looks like we don't use User Voice anymore, so that link is broken. I'm trying to find a replacement, but not sure of that yet.

**Comment by aLifeOnMars** [comment]
Any news on this feature?

**Reply by timtay-microsoft** [reply]
No news yet. It is still a preview-only feature with no plans to make it generally available by the service team.
